wimm.Guardian/EnumArgumentOutOfRangeException.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace wimm.Guardian
+{
+    /// <summary>
+    /// An exception to indicate that an enum arguments values fall outside the set values for that enum
+    /// </summary>
+    public class EnumArgumentOutOfRangeException : Exception
```

Comment:
Any reason not to derive from `ArgumentException`?

Reply:
🚢 it